File: Module.Persist.TPM/Model/TPM/BudgetSubItemClientTree.cs
using Core.Data;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Module.Persist.TPM.Model.TPM
{
    public class BudgetSubItemClientTree : IEntity<Guid>
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }

        public int ClientTreeId { get; set; }
        public virtual ClientTree ClientTree { get; set; }
        public Guid BudgetSubItemId { get; set; }
        public virtual BudgetSubItem BudgetSubItem { get; set; }
    }
}

File: Module.Frontend.TPM/Client/model/tpm/budgetsubitem/BudgetSubItemClientTree.js
Ext.define('App.model.tpm.budgetsubitem.BudgetSubItemClientTree', {
    extend: 'Ext.data.Model',
    idProperty: 'Id',
    breezeEntityType: 'BudgetSubItemClientTree',
    fields: [
        { name: 'Id', hidden: true },
        { name: 'ClientTreeId', hidden: true },
        { name: 'BudgetSubItemId', hidden: true },
        { name: 'ClientTreeFullPathName', type: 'string', isDefault: true, mapping: 'ClientTree.FullPathName', defaultFilterConfig: { valueField: 'FullPathName' } },
        { name: 'ClientTreeObjectId', type: 'int', isDefault: true, mapping: 'ClientTree.ObjectId', defaultFilterConfig: { valueField: 'ObjectId' } }
    ],
    proxy: {
        type: 'breeze',
        resourceName: 'BudgetSubItemClientTrees',
        reader: {
            type: 'json',
            totalProperty: 'inlineCount',
            root: 'results'
        }
    }
});
